src/Request/Source/File.php

```diff
@@ -0,0 +1,122 @@
+<?php
+declare(strict_types=1);
+
+namespace YapepBase\Request\Source;
+
+use YapepBase\Application;
+
+/**
+ * Data class for holding uploaded file data.
+ */
+class File
+{
+    /** @var string */
+    protected $filename;
+
+    /** @var int */
+    protected $sizeInByte;
+
+    /** @var string */
+    protected $temporaryFilePath;
+
+    /** @var string|null */
+    protected $mimeType;
+
+    /** @var int */
+    protected $errorCode;
+
+    public function __construct(string $filename, int $sizeInByte, string $tempFilePath, int $errorCode, ?string $mimeType)
+    {
+        $this->filename          = $filename;
+        $this->sizeInByte        = $sizeInByte;
+        $this->temporaryFilePath = $tempFilePath;
+        $this->mimeType          = $mimeType;
+        $this->errorCode         = $errorCode;
+
+        $this->correctSizeIfNeeded();
+    }
+
+    /**
+     * If the filesize is smaller than 0, we have a file that's bigger than 2GB.
+     */
+    protected function correctSizeIfNeeded(): void
```

Comment:
Is this still needed? This was fixed in PHP for a few years now.

Reply:
Removed